### PR TITLE
Light node/fetcher downloader

### PR DESCRIPTION
**Pull request by jwangdx**
### Description

1. Adapted Dpos related content to fetcher
2. Adapted Dpos related content to les.downloader
3. Updated eth.protocolManager for new Downloader

### CODE REVIEW

File: eth/handler.go [hunk 1]
```diff
@@ -26,6 +26,8 @@ import (
 	"sync/atomic"
 	"time"
 
+	"github.com/DxChainNetwork/godx/consensus/dpos"
```
Comment (xzhangdx) [inline eth/handler.go]: format import package: remove empty line bewteen internal packages

File: eth/protocol.go [hunk 2]
```diff
@@ -21,6 +21,8 @@ import (
 	"io"
 	"math/big"
 
+	"github.com/DxChainNetwork/godx/p2p"
```
Comment (xzhangdx) [inline eth/protocol.go]: format import package: remove empty line bewteen internal packages

File: les/protocol.go [hunk 3]
```diff
@@ -24,6 +24,11 @@ import (
 	"io"
 	"math/big"
 
+	"github.com/DxChainNetwork/godx/light"
+	"github.com/DxChainNetwork/godx/p2p"
+
```
Comment (xzhangdx) [inline les/protocol.go]: format import package: remove empty line bewteen internal packages

File: core/genesis.go [hunk 4]
```diff
@@ -492,3 +543,37 @@ func initGenesisDposContext(stateDB *state.StateDB, g *Genesis, db ethdb.Databas
...
+		return err
+	}
+	if balance := common.PtrBigInt(stateDB.GetBalance(addr)); balance.Cmp(validatorInfo.Deposit) < 0 {
+		return fmt.Errorf("during genesis intialization, validator %x does not have enough balance", addr)
+	}
+	dpos.SetCandidateDeposit(stateDB, addr, validatorInfo.Deposit)
+	dpos.SetFrozenAssets(stateDB, addr, validatorInfo.Deposit)
+	dpos.SetRewardRatioNumerator(stateDB, addr, validatorInfo.RewardRatio)
+	dpos.SetRewardRatioNumeratorLastEpoch(stateDB, addr, validatorInfo.RewardRatio)
+}
```
Comment (xzhangdx) [inline core/genesis.go]: missing return at end of function

File: core/genesis.go [hunk 5]
```diff
@@ -364,6 +372,61 @@ func DefaultGenesisBlock() *Genesis {
 	return g
 }
 
+// DefaultGenesisBlock returns the Ethereum main net genesis block.
```
Comment (xzhangdx) [inline core/genesis.go]: change `DefaultGenesisBlock ` to `TestGenesisBlockWithInitialCandidates ` for this comment

File: eth/downloader/testchain_test.go [hunk 6]
```diff
@@ -18,23 +18,26 @@ package downloader
 
 import (
 	"fmt"
+	"math/big"
+	"sync"
+
+	"github.com/DxChainNetwork/godx/consensus"
```
Comment (xzhangdx) [inline eth/downloader/testchain_test.go]: format import package: remove the empty line between internal packages

File: eth/handler.go (hunk 1 above)
Comment (jwangdx) [reply eth/handler.go]: Thx, fixed.

File: les/protocol.go (hunk 3 above)
Comment (jwangdx) [reply les/protocol.go]: Thx, fixed.

File: core/genesis.go (hunk 4 above)
Comment (jwangdx) [reply core/genesis.go]: Thx, fixed.

File: eth/protocol.go (hunk 2 above)
Comment (jwangdx) [reply eth/protocol.go]: Thx, fixed

File: core/genesis.go (hunk 5 above)
Comment (jwangdx) [reply core/genesis.go]: Thx, fixed.

File: eth/downloader/testchain_test.go (hunk 6 above)
Comment (jwangdx) [reply eth/downloader/testchain_test.go]: Thx, fixed.

**Comment by jwangdx** [comment]
Closed and will be opened in another PR.

**Pull request closed without merging**